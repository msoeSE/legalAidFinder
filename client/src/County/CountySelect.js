import React, { Component } from 'react';
import { Link } from 'react-router';
import PropTypes from 'prop-types';

class CountySelectPage extends Component {
    componentDidMount() {
        this.props.dispatch(fetchCounties());
    }

    chooseCounty = (county) => {
        this.setState({ chosenCounty: county });
    };

    addSelectedCountyClick() {
        addSelectedCounty(this.props.chosenCounty);
    }

    render() {
        return (
            <div>
                <div>
                    <h2 className={`${styles['county-title']}`}>Select the county in Wisconsin which you reside in:</h2>
                </div>
                <CountySelector counties={this.props.counties} handleCountyChange={this.chooseCounty} />
                {<Link to="/categories/" onClick={this.addSelectedCountyClick}>
                    Submit
                </Link>}
            </div>
        );
    }
}

// Actions required to provide data for this component to render in sever side.
CountySelectPage.need = [() => { return fetchCounties(); }];

// Retrieve data from store as props
function mapStateToProps(state) {
    return {
        counties: getCounties(state)
    };
}

CountySelectPage.propTypes = {
    counties: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string.isRequired,
        state: PropTypes.string,
        _id: PropTypes.string.isRequired,
    })).isRequired,
    chosenCounty: PropTypes.string,
    dispatch: PropTypes.func.isRequired,
};

CountySelectPage.contextTypes = {
    router: PropTypes.object,
};

export default connect(mapStateToProps)(CountySelectPage);