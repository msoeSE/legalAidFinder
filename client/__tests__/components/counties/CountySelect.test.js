import configureMockStore from 'redux-mock-store';
import React from 'react';
import { connect } from 'react-redux';
import { shallowWithStore } from 'enzyme-redux';
import { createMockStore } from 'redux-test-utils';
import CountySelect from '../../../src/County/CountySelect';
import {shallow, mount} from 'enzyme';

const mockStore = configureMockStore();

describe('<CountySelect />', function () {
    it('renders without exploding', () => {
        const store = mockStore({chosenCounty: ""});
        expect(
            shallow(
                <CountySelect store={store} />
            ).length).toEqual(1);
    });

    it('has a working state', () =>{
        const expectedState = {chosenCounty: ""};
        const mapStateToProps = (state) => ({
            state,
        });
        const ConnectedComponent = connect(mapStateToProps)(CountySelect);
        const component = shallowWithStore(<ConnectedComponent />, createMockStore(expectedState));
        expect(component.props().state).toBe(expectedState);
    });

    it('can dispatch an action', () => {
        const expectedState = {chosenCounty: ""};
        const action = {
            type: 'CHOOSE_COUNTY',
            payload: "Adams"
        };
        const mapDispatchToProps = (dispatch) => ({
            dispatchProp() {
                dispatch(action);
            },
        });
        const store = createMockStore(expectedState);

        const ConnectedComponent = connect(undefined, mapDispatchToProps)(CountySelect);
        const component = shallowWithStore(<ConnectedComponent />, store);
        component.props().dispatchProp();
        expect(store.isActionDispatched(action)).toBe(true);
    });
});