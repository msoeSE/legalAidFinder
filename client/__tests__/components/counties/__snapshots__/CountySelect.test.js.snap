// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# CountySelect page 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route
    render={[Function]}
  />,
  "nodes": Array [
    <Route
      render={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <withRouter(Connect(CountySelect)) />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Route
          render={[Function]}
        />,
        "_debugID": 4,
        "_renderedOutput": <Route
          render={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(Connect(CountySelect)) />,
}
`;

exports[`# CountySelect page wrapper 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route
    render={[Function]}
  />,
  "nodes": Array [
    <Route
      render={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <withRouter(Connect(CountySelect))
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Route
          render={[Function]}
        />,
        "_debugID": 2,
        "_renderedOutput": <Route
          render={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(Connect(CountySelect))
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
}
`;

exports[`CountySelect page wrapper 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Route
    render={[Function]}
  />,
  "nodes": Array [
    <Route
      render={[Function]}
    />,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <withRouter(Connect(CountySelect))
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Route
          render={[Function]}
        />,
        "_debugID": 2,
        "_renderedOutput": <Route
          render={[Function]}
        />,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <withRouter(Connect(CountySelect))
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
}
`;
